prow/cmd/gerrit/main_test.go

```diff
@@ -71,4 +90,32 @@ func TestSyncTime(t *testing.T) {
 	if actual := st.Current(); !actual.After(cur) {
 		t.Errorf("Update(%v) did not move current value to after %v, got %v", later, cur, actual)
 	}
+
+	expected := later.Truncate(time.Second)
+	st = syncTime{
+		path:   path,
+		opener: open,
+		ctx:    ctx,
+	}
+	if err := st.init(); err != nil {
+		t.Fatalf("Failed init: %v", err)
+	}
+	if actual := st.Current(); !actual.Equal(expected) {
+		t.Errorf("init() failed to reload %v, got %v", expected, actual)
+	}
+
+	st = syncTime{
+		path: path,
+		opener: fakeOpener{
+			opener:   open,
+			notFound: true,
+		},
+		ctx: ctx,
+	}
+	if err := st.init(); err != nil {
+		t.Fatalf("Failed init: %v", err)
+	}
+	if cur := st.Current(); now.After(cur) {
```

Comment:
hummmmmm what is this trying to test? I think we should compare against `expected` which is what currently stored in the `path`, but invalidated by `notFound: true`?

Reply:
This is testing that if GCS returns a 404 we respond the same way as if we do not find a local file (aka fail to time.Now)

Reply:
I think maybe you want to `st.Update(earlier)` first?

Or the old `now` will always be before `cur` (new now, failure mode in 404) or `later` (whatever stored in `path`)?

Reply:
Maybe this is more clear